### DISCUSSION TITLE
Failed to read the 'localStorage' property from 'Window': Access is denied for this document.

### DISCUSSION THREAD

**Discussion by meotimdihia** [pmndrs/jotai#1591] [Q&A] [answered]
I am getting this error: `Failed to read the 'localStorage' property from 'Window': Access is denied for this document.`
```ts
export const storedAtom = atomWithStorage(
  'colored',
  {}
)
```

Do we have a way to catch this error if the user disables `localStorage` for the website?

**Comment by dai-shi** [comment, marked as answer]
You can pass a custom storage instead of the default one.
https://github.com/pmndrs/jotai/blob/bfd0af0bea8264430480f3cb4956d54e8f6ddf5e/src/utils/atomWithStorage.ts#L98-L102
Using `createJSONStorage` is even optional.
